filebeat/input/file/state.go

```diff
@@ -31,6 +32,16 @@ func NewState(fileInfo os.FileInfo, path string) State {
 	}
 }
 
+// String returns a unique id for the state as a string
+func (s *State) String() string {
+	return s.FileStateOS.String()
+}
+
+// IsEqual compares the state to an other state supporing stringer based on the unique string
+func (s *State) IsEqual(c fmt.Stringer) bool {
+	return s.String() == c.String()
```

Comment:
I initially had a `ID()` function for comparison. I switched it to `String()` because the `Stringer` interface already exist. But perhaps have a local interface with `ID() string` is a better choice? +1 on storing it locally and not generating it every time, as it should not change over time.